smack-integration-test/src/main/java/org/jivesoftware/smackx/muc/AbstractMultiUserChatIntegrationTest.java

```diff
@@ -55,10 +55,30 @@ public AbstractMultiUserChatIntegrationTest(SmackIntegrationTestEnvironment envi
 
         List<DomainBareJid> services = mucManagerOne.getMucServiceDomains();
         if (services.isEmpty()) {
-            throw new TestNotPossibleException("No MUC (XEP-45) service found");
+            throw new TestNotPossibleException("No MUC (XEP-0045) service found");
         }
 
-        mucService = services.get(0);
+        DomainBareJid needle = null;
+        for (final DomainBareJid service : services) {
+            try {
+                final String roomNameLocal = String.join("-", "smack-inttest-abstract", testRunId, StringUtils.insecureRandomString(6));
+                final EntityBareJid mucAddress = JidCreate.entityBareFrom(Localpart.from(roomNameLocal), service.getDomain());
+                final MultiUserChat multiUserChat = mucManagerOne.getMultiUserChat(mucAddress);
+                createMuc(multiUserChat, Resourcepart.from("test"));
+                tryDestroy(multiUserChat);
```

Comment:
I'd rather put this tryDestroy() in a `finally` block and let it throw instead of catching its exceptions.

Reply:
I've moved that to a `finally` block, to be executed only when there was a successful creation.